src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -348,9 +351,42 @@ void PCRPrimerDesignForDNAAssemblyOPWidget::sl_onFindTaskFinished() {
     CHECK(pcrTask->isFinished(), );
     productsTable->setCurrentProducts(pcrTask->getResults(), annDnaView);
     createResultAnnotations();
+    backboneSequence = pcrTask->getBackboneSequence();
+    lastRunSettings = pcrTask->getSettings();
     pcrTask = nullptr;
 }
 
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_extractProduct() {
+    ADVSequenceObjectContext *sequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(NULL != sequenceContext, L10N::nullPointerError("Sequence Context"), );
+    U2SequenceObject *sequenceObject = sequenceContext->getSequenceObject();
+    SAFE_POINT(NULL != sequenceObject, L10N::nullPointerError("Sequence Object"), );
+    ExtractPrimerTaskSettings settings;
+    settings.sequenceRef = sequenceContext->getSequenceRef();
+    GUrl sequenceURL = sequenceContext->getSequenceObject()->getDocument()->getURL();
+    if (sequenceURL.getType() == GUrl_File) {
+        settings.originalSequenceFileName = QFileInfo(sequenceURL.getURLString()).baseName();
+    } else {
+        settings.originalSequenceFileName = sequenceContext->getSequenceObject()->getSequenceName();
+    }
+    if (lastRunSettings.insertTo == PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone5) {
+        settings.direction = ExtractPrimerTaskSettings::FiveToFive;
+        settings.backboneLength = lastRunSettings.bachbone5Length;
+    } else {
+        CHECK(lastRunSettings.insertTo == PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone3, )
```

Comment:
Hmmm... Don't get why this check is needed. Please, add a comment

Reply:
Just to be sure what one of options is selected. I can remove this if you found it useless.